subprojects/luajit/src/host/meson.build

```diff
@@ -11,25 +11,33 @@ endif
 dynasm_dasc = files('../vm_@0@.dasc'.format(dynasm_arch))
 dasm = [minilua, files('../../dynasm/dynasm.lua')]
 
-# BUG: meson does not resolve paths correctly for subprojects
-hpre = '#include "@0@/../lj_arch.h"'.format(meson.current_source_dir())
-
-checkdefs = [
-    ['LJ_ARCH_BITS', '64', ['-D', 'P64']],
-    ['LJ_HASJIT', '1', ['-D', 'JIT']],
-    ['LJ_HASFFI', '1', ['-D', 'FFI']],
-    ['LJ_DUALNUM', '1', ['-D', 'DUALNUM']],
-]
-
-if host_machine.cpu_family() == 'x86'
-    checkdefs += ['__SSE2__', '1', ['-D', 'SSE']]
-endif
-
-foreach def: checkdefs
-    if cc.get_define(def[0], prefix: hpre) == def[1]
-        dasm += def[2]
+if cc.get_id() == 'msvc'
+    # the cl.exe preprocessor seemingly removes/expands macros, so hardcode flags
+    dasm += ['-D', 'WIN', '-D', 'JIT', '-D', 'FFI']
+    if host_machine.cpu_family() == 'x86_64'
+        dasm += ['-D', 'P64']
+    endif
+else
+    # BUG: meson does not resolve paths correctly for subprojects
```

Comment:
Is there an issue open about this? I'd check lachs0r's old issues in case he opened one and this is now fixed, and otherwise we should probably open one ourselves.

Reply:
I'm not familiar with the bug myself, I just kept the old code as-is on non-Windows.

Reply:
Yeah, I figured. I'd like to get an issue open or change this back though, so I'll take a look.

Reply:
I can't figure out what was meant here, so I'll just leave it for now. The note was present in the very first commit and no issue was opened.